lib/AST/TypeCheckRequests.cpp

```diff
@@ -1501,3 +1501,23 @@ void swift::simple_display(llvm::raw_ostream &out,
   out << "implicit import of ";
   simple_display(out, import.Module);
 }
+
+//----------------------------------------------------------------------------//
+// ResolveTypeRequest computation.
+//----------------------------------------------------------------------------//
+
+void ResolveTypeRequest::noteCycleStep(DiagnosticEngine &diags) const {
+  auto *repr = std::get<1>(getStorage());
+  diags.diagnose(repr->getLoc(), diag::circular_type_resolution_note, repr);
+}
+
+void swift::simple_display(llvm::raw_ostream &out,
+                           const TypeResolution *resolution) {
+  out << "while resolving type ";
```

Comment:
Should this also print the type?

Reply:
This definition of `simple_display` is a layering violation. `TypeResolution` lives in libSema.

Reply:
Not much more I can do here, I'm afraid. I have no ability to access members.

Reply:
simple_display will fire on the adjacent typerepr, so this should render just fine.